workerpool/example/consumer.go

```diff
@@ -11,13 +13,26 @@ import (
 type consumer struct {
 	log zerolog.Logger
 
+	tid                 workerpool.ThreadID
 	queue               workerpool.SubmissionQueue
 	completed           uint64
 	workCompletedReal   uint64
 	workCompletedCanary uint64
 }
 
+// rampUp linearly in increasing increments of the 5 seconds. This function is
+// only called once when each consumer is first Run.
+func (c *consumer) rampUp() {
+	rampUpDur := time.Duration((c.tid)*5) * time.Second
```

Comment:
Push this value into the `consumerFactory` and then have the consumer factory set this value in the `consumer`.

Reply:
Ah right, that'll make it more configurable.

Reply:
I forgot that this calculation requires `ThreadID` at `Run` time. Should we pass `ThreadID` into `ConsumerFactory.New` and store the `ThreadID` on the `Consumer` struct as a field?

The changed interfaces would be...

```go
type ConsumerFactory interface {
	New(ThreadID, SubmissionQueue) (Consumer, error)
}

type Consumer interface {
	Run(context.Context) error
}
```

That would let me setup things like this at construction from within the factory.